data/scripts/update_ip_country_data.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python3
+# COPYRIGHT (C) 2024 Nicotine+ Contributors
+#
+# GNU GENERAL PUBLIC LICENSE
+#    Version 3, 29 June 2007
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import csv
+import io
+import os
+import time
+import urllib.request
+import zipfile
+
+
+def parse_ip_country_data():
+    """Download and parse IP2Location country data."""
+
+    country_data_url = "https://download.ip2location.com/lite/IP2LOCATION-LITE-DB1.CSV.ZIP"
+    copyright_prefix = b"Copyright (c)"
+    expected_copyright_author = b"Hexasoft Development"
+    expected_license = b"Creative Commons Attribution-ShareAlike 4.0 International"
+    max_ipv4_range = 4294967295
+    file_path, _headers = urllib.request.urlretrieve(country_data_url)
```

Comment:
Perhaps there are data in `headers` that can be used to verify the authenticity of the source remote host?

Ideally the author's public key could be parsed to compare it for a match, albeit that might be a bit overkill... 

Perhaps something more trivial can be checked to satisfy the operator of the script that the domain is still under the same ownership organisation or else bail, or that the file is within a sane size that would be expected, etc.

Reply:
I added a size check. Anything else should be caught by the parser.